mlflow/deployments/utils.py

```diff
@@ -15,3 +19,44 @@ def parse_target_uri(target_uri):
             + "Deployment URIs must be of the form 'target' or 'target:/suffix'"
         )
     return parsed.scheme
+
+
+def _is_valid_uri(uri: str) -> bool:
+    """
+    Evaluates the basic structure of a provided gateway uri to determine if the scheme and
+    netloc are provided
+    """
+    if uri == "databricks":
+        return True
+    try:
+        parsed = urlparse(uri)
+        return parsed.scheme == "databricks" or all([parsed.scheme, parsed.netloc])
+    except ValueError:
+        return False
+
+
+def assemble_uri_path(paths: List[str]) -> str:
+    """
+    Assemble a correct URI path from a list of path parts.
+
+    :param paths: A list of strings representing parts of a URI path.
+    :return: A string representing the complete assembled URI path.
+    """
+    stripped_paths = [path.strip("/").lstrip("/") for path in paths if path]
```

Comment:
```suggestion
    stripped_paths = [path.strip("/") for path in paths if path]
```

`lstrip` here seems useless.

Reply:
@harupy are you sure we don't have a similar util elsewhere, e.g. in `mlflow.utils`? Seems like something we have elsewhere

Reply:
let me search.

Reply:
Can we remove this function and use `posixpath.normpath(posixpath.join(x))`?

Reply:
Sure, but I'm shocked that we don't have a util / similar logic elsewhere

Reply:
We have `_join_posixpaths_and_append_absolute_suffixes` here:

https://github.com/mlflow/mlflow/blob/a1db4c34be6d3f93878ed326baa478d67e3984d4/mlflow/utils/uri.py#L298

This is the closest to what we need.

Reply:
I prefer using `posixpath.normpath(posixpath.join(x))` for now tho.